react-carousel-component/src/carousel.jsx

```diff
@@ -0,0 +1,95 @@
+import React, { useState, useEffect } from 'react';
+
+const CarouPics = ({ index, isClicked, setIsClicked, data }) => {
+  return (
+    <div key={'container' + index}>
+      <img key={'image' + index} src={data} className={isClicked === index ? 'image' : 'image hidden'} />
+    </div>
+  );
+};
+
+const LeftChevron = ({ onClickFnLeft }) => {
+  return <i className='fa-solid fa-chevron-left chevron-adjust' onClick={onClickFnLeft}></i>;
+};
+
+const RightChevron = ({ onClickFnRight }) => {
+  return <i className='fa-solid fa-chevron-right chevron-adjust' onClick={onClickFnRight}></i>;
+};
+
+function Carousel(props) {
+  const [isClicked, setIsClicked] = useState(0);
+  const Arraylength = props.images.length;
+
+  const onClickFnIcon = index => {
+    setIsClicked(prestate => {
+      clearInterval(intervalID);
+      carousal();
+      return index;
+    });
+  };
+  const icons = props.images.map((icon, index) => {
+    return <i key={index} className={isClicked === index ? 'fa-solid fa-circle icon-adjust' : 'fa-regular fa-circle icon-adjust'}
+    onClick={() => { onClickFnIcon(index); }}></i>;
+  });
+
+  let intervalID;
```

Comment:
If you store/set this interval id in state like isClicked, then i will approve it. this `let` variable will not be used by your effect cleanup if your user goes to a different screen, then they will get a memory leak.